CONTRIBUTING.md

```diff
@@ -1,150 +1,249 @@
+    - [Clone the git repository](#clone-the-git-repository)
+    - [Run the tests locally](#run-the-tests-locally)
+  - [Requirements for Pull Requests](#requirements-for-pull-requests)
+    - [Release notes](#release-notes)
+  - [CI (Continuous Integration)](#ci-continuous-integration)
+  - [Working from Github forks](#working-from-github-forks)
+  - [Writing tests](#writing-tests)
+    - [Unit test](#unit-test)
+    - [Integration test](#integration-test)
+    - [End to End (e2e) test](#end-to-end-e2e-test)
+  - [Contributor Licence Agreement (CLA)](#contributor-licence-agreement-cla)
+
+## Code of Conduct
+
+This project and everyone participating in it is governed by our [Code of Conduct](code_of_conduct.txt).
+By participating, you are expected to uphold this code. Please report unacceptable behavior to haystack@deepset.ai.
+
+## I Have a Question
+
+> [!TIP]
+> If you want to ask a question, we assume that you have read the available [Documentation](https://docs.haystack.deepset.ai/v2.0/docs/intro).
+
+Before you ask a question, it is best to search for existing [Issues](/issues) that might help you. In case you have
+found a suitable issue and still need clarification, you can write your question in this issue. It is also advisable to
+search the internet for answers first.
+
+If you then still feel the need to ask a question and need clarification, you can use one of our
+[Community Channels](https://haystack.deepset.ai/community), Discord in particular is often very helpful.
+
+## Reporting Bugs
+
+### Before Submitting a Bug Report
+
+A good bug report shouldn't leave others needing to chase you up for more information. Therefore, we ask you to
+investigate carefully, collect information and describe the issue in detail in your report. Please complete the
+following steps in advance to help us fix any potential bug as fast as possible.
+
+- Make sure that you are using the latest version.
+- Determine if your bug is really a bug and not an error on your side, for example using incompatible versions.
+  Make sure that you have read the [documentation](https://docs.haystack.deepset.ai/v2.0/docs/intro). If you are looking
+  for support, you might want to check [this section](#i-have-a-question).
+- To see if other users have experienced (and potentially already solved) the same issue you are having, check if there
+  is not already a bug report existing for your bug or error in the [bug tracker](/issues).
+- Also make sure to search the internet (including Stack Overflow) to see if users outside of the GitHub community have
+  discussed the issue.
+- Collect information about the bug:
+  - OS, Platform and Version (Windows, Linux, macOS, x86, ARM)
+  - Version of Haystack and the integrations you're using
+  - Possibly your input and the output
+  - If you can reliably reproduce the issue, a snippet of code we can use
+
+### How Do I Submit a Good Bug Report?
```

Comment:
should we mention somewhere that if the bug refers to a core integration, it should be submitted in the appropriate repo?

Reply:
I would keep it simple, it's already unlikely anybody will go through this doc line by line 😅 - I would rather put this note in the README if integrations issues become a problem.